core/src/test/scala/org/apache/spark/scheduler/BlacklistIntegrationSuite.scala

```diff
@@ -51,37 +54,67 @@ class BlacklistIntegrationSuite extends SchedulerIntegrationSuite[MultiExecutorM
     assertDataStructuresEmpty(noFailure = false)
   }
 
-  // even with the blacklist turned on, if maxTaskFailures is not more than the number
-  // of executors on the bad node, then locality preferences will lead to us cycling through
-  // the executors on the bad node, and still failing the job
+  // even with the blacklist turned on, bad configs can lead to job failure.  To survive one
+  // bad node, you need to make sure that
+  // maxTaskFailures > min(spark.blacklist.task.maxTaskAttemptsPerNode, nExecutorsPerHost)
   testScheduler(
     "With blacklist on, job will still fail if there are too many bad executors on bad host",
     extraConfs = Seq(
-      // set this to something much longer than the test duration so that executors don't get
-      // removed from the blacklist during the test
-      ("spark.scheduler.executorTaskBlacklistTime", "10000000")
+      config.BLACKLIST_ENABLED.key -> "true",
+      config.MAX_TASK_ATTEMPTS_PER_NODE.key -> "5",
+      config.MAX_TASK_FAILURES.key -> "4",
+      "spark.testing.nHosts" -> "2",
+      "spark.testing.nExecutorsPerHost" -> "5",
+      "spark.testing.nCoresPerExecutor" -> "10",
+      // Blacklisting will normally immediately complain that this config is invalid -- the point
+      // of this test is to expose that the configuration is unsafe, so skip the validation.
+      "spark.blacklist.testing.skipValidation" -> "true"
```

Comment:
there are separate checks for the validation (`BlacklistTrackerSuite` "check blacklist configuration invariants") so I'll just delete this test.